src/plugins/data/common/search/search_source/search_source.ts

```diff
@@ -412,6 +455,50 @@ export class SearchSource {
           const dataFrameResponse = response as IDataFrameResponse;
           await this.setDataFrame(dataFrameResponse.body as IDataFrame);
           return onResponse(searchRequest, convertResult(response as IDataFrameResponse));
+        } else if ((response as IDataFrameResponse).type === DATA_FRAME_TYPES.POLLING) {
+          const dataFrameResponse = response as IDataFrameResponse;
+          const responseQueryId = (dataFrameResponse.body as IDataFrame).meta?.queryId;
+          const responseSessionId = (dataFrameResponse.body as IDataFrame).meta?.sessionId;
+          // if there wasn't a session before, set current datasource as session
+          // TODO: MQL update with selected datasource
+          if (!sessionId || (sessionId && sessionId !== responseSessionId)) {
+            this.setSession('mys3', responseSessionId);
+          }
+          const queryIdRequest: ISearchRequestParams = {
```

Comment:
i think this logic should exist in a dataframe polling or in the search strategy can handle this logic since it will get the search request anyways.

Reply:
Was having issues passing this in, since we need the `setSession` function from `SearchService`. Is there a way to call the session cache from `SearchService` somewhere in the search interceptor or strategy?